### ISSUE TITLE
Implement Monte-Carlo Dropout (uncertainty evaluation)

### ISSUE THREAD

**Issue by florian-huber** [matchms/ms2deepscore#53]
The `SiameseModel` is trained using dropout, which helps to prevent overfitting. But it also is there to help assess the uncertainty of the model by using Monte-Carlo Dropout. 

Right now this lives only in a few notebooks, so it should become part of the main code.

**Comment by florian-huber** [comment]
One trade-off issue here is the following:
Performance-wise, the main computational cost of our scores are the 2 first steps: `(1) Spectrum -> BinnedSpectrum`, and `(2) BinnedSpectrum -> embedding`. So ideally we do not want to duplicate such steps.
However, if we compute scores between a high number of spectrums that can get complicate memory-wise. Say, we compute the scores for all pairs between 10,000 spectrums, than we need to compute 10,000 embeddings and run `cosine_similarity_matrix`. That's actually still fine.
But what if we increase the number further, and/or increase `n_ensemble` (code example above) so that we compute `n_ensemble` different embeddings for every spectrum...

One alternative is to break it up into sub-parts, but then we have to calculate some embeddings several times (hence add computational cost).

Another way to go might be, to first compute the most expensive part that can still be used for all further steps for all spectrums.
- The first step `(1) Spectrum -> BinnedSpectrum` can easily be done for all spectrums.

- The second step could be splitted: `(2a) BinnedSpectrum -> 1.layer embedding` `(2b) 1.layer embedding -> output embedding`    
The reason for this would be that the 1st dense layer does not use `dropout` so it will always give the same result and at the same time compress the input data quiet a bit!

**Comment by florian-huber** [comment]
Spec2Vec has been added to GNPS (release 27)!
--> https://github.com/CCMS-UCSD/GNPS_Workflows/releases

**Comment by florian-huber** [comment]
Done with #65
